testsuite/tests/basic/patmatch.ml

```diff
@@ -149,8 +149,8 @@ let () =
 
 (* No string sharing PR#6322 *)
 let test x = match x with
-  | true -> "a"
-  | false -> "a"
+  | true -> Bytes.of_string "a"
+  | false -> Bytes.of_string "a"
```

Comment:
In which case, shall I split that test out to a separate file and do a similar shim to https://github.com/ocaml/ocaml/commit/fa244ddaea2aa258fc8f4e38ac0768affd2f31c5 to put a dummy ML file for `-safe-string` mode?